srcalc/src/main/java/gov/va/med/srcalc/web/controller/CalculationController.java

```diff
@@ -44,6 +44,39 @@ public ModelAndView startNewCalculation(
         mav.addObject("specialties", fCalculationService.getValidSpecialties());
         return mav;
     }
+
+    @RequestMapping(value = "/newCalc", method = RequestMethod.GET)
+    public ModelAndView startNewCalculation(
+            @RequestParam(value = "patientDfn") final int patientDfn,
+            final HttpSession session)
+    {
+        // If there is a calculation already in the session, ask the user
+        // if they wish to override the in-progress calculation.
+        try
+        {
+            final CalculationSession calcSession = SrcalcSession.getCalculationSession(session);
+            final Calculation calc = calcSession.getCalculation();
+            final ModelAndView mav = new ModelAndView("redirect:/confirmNewCalc");
+            mav.addObject("calculation", calc);
+            return mav;
+        }
+        catch(final IllegalStateException e)
```

Comment:
Hmmmm, I think this violates using exceptions for exceptional circumstances. I think we should just add a method to SrcalcSession to check if there is already a CalculationSession.